packages/flutter_tools/lib/src/cache.dart

```diff
@@ -248,26 +282,52 @@ abstract class CachedArtifact {
   /// starting from scratch.
   final List<File> _downloadedFiles = <File>[];
 
-  bool isUpToDate() {
-    if (!location.existsSync())
-      return false;
-    if (version != cache.getStampFor(name))
-      return false;
-    return isUpToDateInner();
+  UpdateResult isUpToDate({
+    BuildMode buildMode,
```

Comment:
`buildMode` and `targetPlatform` should be properties of the `CachedArtifact` itself, rather than exist on the update checks, no?

Reply:
These are passed through so that we can filter down the set of artifacts based on the current known configurations.